### ISSUE TITLE
I can't get the same results as you.

### ISSUE THREAD

**Issue by Yrchuan** [burness/tensorflow-101#10]
Hi, I downloaded your code and then run it in linux server, I didn't change the code , just modifed some function parameter name because of the issue of  tensorflow versions.
I ran train.py first and successfully generated checkpoint documents.
information printed:

> cur epoch 49 update l_d step 84182, loss_disc 0.171038478613, loss_gen 7.14091253281
> cur epoch 50 update l_d step 85900, loss_disc 0.114559516311, loss_gen 8.29001522064
> W tensorflow/core/kernels/queue_base.cc:294] _2_test/input_producer/fraction_of_4096_full/fraction_of_4096_full: Skipping cancelled enqueue attempt with queue not closed
> W tensorflow/core/kernels/queue_base.cc:294] _4_valid_1/random_shuffle_queue: Skipping cancelled enqueue attempt with queue not closed
> W tensorflow/core/kernels/queue_base.cc:294] _1_valid/input_producer/fraction_of_4096_full/fraction_of_4096_full: Skipping cancelled enqueue attempt with queue not closed
> W tensorflow/core/kernels/queue_base.cc:294] _3_test_1/random_shuffle_queue: Skipping cancelled enqueue attempt with queue not closed
> W tensorflow/core/kernels/queue_base.cc:294] _0_train/input_producer/fraction_of_4096_full/fraction_of_4096_full: Skipping cancelled enqueue attempt with queue not closed

 
Then I ran generate.py, 
information printed:

> (100, 50)
> Tensor("generator/fully_connected/Relu:0", shape=(100, 1024), dtype=float32)
> gen (100, 14, 14, 64)
> [2 2 4 3 5 6 4 1 2 7 1 2 5 9 7 7 7 2 5 9 4 3 1 4 7 1 3 3 2 1 9 0 3 8 1 0 5
>  2 5 5 5 5 1 0 5 8 4 4 8 2 0 6 3 7 8 6 9 7 0 8 7 4 4 4 1 0 1 2 5 0 5 8 7 4
>  1 0 5 4 2 0 8 4 3 5 9 9 8 2 4 3 8 2 7 3 5 9 0 2 5 4]

but I can only generate images of number 1 and 7, I ran the code several times but did not get perfect results like you yet.
Sorry I can't attach image here, I run the code by GPU, can you tell me why? thank you!

**Comment by burness** [comment]
@Yrchuan It seems that the loss_gen is too big, can you attach the all log with the code ?

**Comment by Yrchuan** [comment]
**# Right, thanks for point it out, maybe I should change the code to the form of w-gan/lsgan to get a better result. This is all the logs of training process:**
wsgao@DB16:/DATA2/data/wsgao/GAN/stone/Info-GAN$ CUDA_VISIBLE_DEVICES=2 python train.py 
I tensorflow/stream_executor/dso_loader.cc:125] successfully opened CUDA library libcublas.so.7.5 locally
I tensorflow/stream_executor/dso_loader.cc:125] successfully opened CUDA library libcudnn.so.5 locally
I tensorflow/stream_executor/dso_loader.cc:125] successfully opened CUDA library libcufft.so.7.5 locally
I tensorflow/stream_executor/dso_loader.cc:125] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:125] successfully opened CUDA library libcurand.so.7.5 locally
W tensorflow/core/platform/cpu_feature_guard.cc:95] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:95] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:95] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:95] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
Extracting ./asset/data/mnist/train-images-idx3-ubyte.gz
Extracting ./asset/data/mnist/train-labels-idx1-ubyte.gz
Extracting ./asset/data/mnist/t10k-images-idx3-ubyte.gz
Extracting ./asset/data/mnist/t10k-labels-idx1-ubyte.gz
WARNING:tensorflow:From train.py:51: concat (from tensorflow.python.ops.array_ops) is deprecated and will be removed after 2016-12-14.
Instructions for updating:
This op will be removed after the deprecation date. Please switch to tf.concat_v2().
WARNING:tensorflow:From train.py:66: concat (from tensorflow.python.ops.array_ops) is deprecated and will be removed after 2016-12-14.
Instructions for updating:
This op will be removed after the deprecation date. Please switch to tf.concat_v2().
(32, 50)
Tensor("generator/fully_connected/Relu:0", shape=(32, 1024), dtype=float32)
gen (32, 14, 14, 64)
False
(32, 28, 28, 1)
True
(32, 28, 28, 1)
loss_d ()
name: "Adam_1"
op: "AssignAdd"
input: "global_step_1"
input: "Adam_1/value"
attr {
  key: "T"
  value {
    type: DT_INT32
  }
}
attr {
  key: "_class"
  value {
    list {
      s: "loc:@global_step_1"
    }
  }
}
attr {
  key: "use_locking"
  value {
    b: false
  }
}

I tensorflow/core/common_runtime/gpu/gpu_device.cc:885] Found device 0 with properties: 
name: GeForce GTX TITAN X
major: 5 minor: 2 memoryClockRate (GHz) 1.076
pciBusID 0000:06:00.0
Total memory: 11.92GiB
Free memory: 4.68GiB
I tensorflow/core/common_runtime/gpu/gpu_device.cc:906] DMA: 0 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:916] 0:   Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:975] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX TITAN X, pci bus id: 0000:06:00.0)
cur epoch 1 update l_d step 1718, loss_disc 0.184011131525, loss_gen 3.74163198471
cur epoch 2 update l_d step 3436, loss_disc 0.246123626828, loss_gen 2.06742095947
cur epoch 3 update l_d step 5154, loss_disc 0.430865734816, loss_gen 1.58519005775
cur epoch 4 update l_d step 6872, loss_disc 0.36340919137, loss_gen 1.69908475876
cur epoch 5 update l_d step 8590, loss_disc 0.246035709977, loss_gen 1.22626888752
cur epoch 6 update l_d step 10308, loss_disc 0.343908458948, loss_gen 2.31579780579
cur epoch 7 update l_d step 12026, loss_disc 0.19682008028, loss_gen 3.06841444969
cur epoch 8 update l_d step 13744, loss_disc 0.247492313385, loss_gen 2.8691072464
cur epoch 9 update l_d step 15462, loss_disc 0.179384827614, loss_gen 3.2870092392
cur epoch 10 update l_d step 17180, loss_disc 0.355320453644, loss_gen 2.94449567795
cur epoch 11 update l_d step 18898, loss_disc 0.254115283489, loss_gen 2.13446235657
cur epoch 12 update l_d step 20616, loss_disc 0.0822585970163, loss_gen 4.00843048096
cur epoch 13 update l_d step 22334, loss_disc 0.300154447556, loss_gen 2.1641702652
cur epoch 14 update l_d step 24052, loss_disc 0.124102294445, loss_gen 2.52932453156
cur epoch 15 update l_d step 25770, loss_disc 0.145584240556, loss_gen 2.08194637299
cur epoch 16 update l_d step 27488, loss_disc 0.266582250595, loss_gen 3.36353373528
cur epoch 17 update l_d step 29206, loss_disc 0.115360692143, loss_gen 4.53943586349
cur epoch 18 update l_d step 30924, loss_disc 0.134285658598, loss_gen 5.37686681747
cur epoch 19 update l_d step 32642, loss_disc 0.161394014955, loss_gen 5.00802612305
cur epoch 20 update l_d step 34360, loss_disc 0.0428995080292, loss_gen 6.33857631683
cur epoch 21 update l_d step 36078, loss_disc 0.104932934046, loss_gen 4.18886613846
cur epoch 22 update l_d step 37796, loss_disc 0.138710692525, loss_gen 3.98987722397
cur epoch 23 update l_d step 39514, loss_disc 0.0770852267742, loss_gen 4.59940338135
cur epoch 24 update l_d step 41232, loss_disc 0.133578598499, loss_gen 5.04696559906
cur epoch 25 update l_d step 42950, loss_disc 0.0761810541153, loss_gen 5.3115811348
cur epoch 26 update l_d step 44668, loss_disc 0.142309606075, loss_gen 7.1889257431
cur epoch 27 update l_d step 46386, loss_disc 0.128257483244, loss_gen 6.07293605804
cur epoch 28 update l_d step 48104, loss_disc 0.0910914614797, loss_gen 5.4206700325
cur epoch 29 update l_d step 49822, loss_disc 0.0965009331703, loss_gen 5.24533081055
cur epoch 30 update l_d step 51540, loss_disc 0.147539168596, loss_gen 6.31976747513
cur epoch 31 update l_d step 53258, loss_disc 0.106248736382, loss_gen 5.66121816635
cur epoch 32 update l_d step 54976, loss_disc 0.152224585414, loss_gen 5.86847305298
cur epoch 33 update l_d step 56694, loss_disc 0.0672409385443, loss_gen 5.11194801331
cur epoch 34 update l_d step 58412, loss_disc 0.0990980789065, loss_gen 7.75625133514
cur epoch 35 update l_d step 60130, loss_disc 0.179215043783, loss_gen 5.17799520493
cur epoch 36 update l_d step 61848, loss_disc 0.148511618376, loss_gen 6.85683107376
cur epoch 37 update l_d step 63566, loss_disc 0.186734497547, loss_gen 5.82958459854
cur epoch 38 update l_d step 65284, loss_disc 0.0867246687412, loss_gen 5.78038215637
cur epoch 39 update l_d step 67002, loss_disc 0.0907433331013, loss_gen 6.25569152832
cur epoch 40 update l_d step 68720, loss_disc 0.0818254351616, loss_gen 4.83339071274
cur epoch 41 update l_d step 70438, loss_disc 0.143496856093, loss_gen 6.08242893219
cur epoch 42 update l_d step 72156, loss_disc 0.13304027915, loss_gen 6.88972330093
cur epoch 43 update l_d step 73874, loss_disc 0.103976517916, loss_gen 5.94063997269
cur epoch 44 update l_d step 75592, loss_disc 0.110978648067, loss_gen 5.91351699829
cur epoch 45 update l_d step 77310, loss_disc 0.181845024228, loss_gen 5.75684833527
cur epoch 46 update l_d step 79028, loss_disc 0.110556051135, loss_gen 6.83772754669
cur epoch 47 update l_d step 80746, loss_disc 0.126977160573, loss_gen 6.33687829971
cur epoch 48 update l_d step 82464, loss_disc 0.0272928625345, loss_gen 8.01826953888
cur epoch 49 update l_d step 84182, loss_disc 0.171038478613, loss_gen 7.14091253281
cur epoch 50 update l_d step 85900, loss_disc 0.114559516311, loss_gen 8.29001522064
W tensorflow/core/kernels/queue_base.cc:294] _2_test/input_producer/fraction_of_4096_full/fraction_of_4096_full: Skipping cancelled enqueue attempt with queue not closed
W tensorflow/core/kernels/queue_base.cc:294] _4_valid_1/random_shuffle_queue: Skipping cancelled enqueue attempt with queue not closed
W tensorflow/core/kernels/queue_base.cc:294] _1_valid/input_producer/fraction_of_4096_full/fraction_of_4096_full: Skipping cancelled enqueue attempt with queue not closed
W tensorflow/core/kernels/queue_base.cc:294] _3_test_1/random_shuffle_queue: Skipping cancelled enqueue attempt with queue not closed
W tensorflow/core/kernels/queue_base.cc:294] _0_train/input_producer/fraction_of_4096_full/fraction_of_4096_full: Skipping cancelled enqueue attempt with queue not closed
wsgao@DB16:/DATA2/data/wsgao/GAN/stone/Info-GAN$ python generate.py 

**As for the code, I only changed the blacken lines**
 -*- coding: utf-8 -*-
import tensorflow as tf
import numpy as np
import logging
from mnist import Mnist
import tensorflow.contrib.slim as slim
from generator import generator
from discriminator import discriminator
from optimizer import optim
from queue_context import queue_context
import os
import sys


_logger = tf.logging._logger
_logger.setLevel(0)


#
# hyper parameters
#

batch_size = 32   # batch size
cat_dim = 10  # total categorical factor
con_dim = 2  # total continuous factor
rand_dim = 38  
num_epochs = 30
debug_max_steps = 1000
save_epoch = 5
max_epochs = 50

#
# inputs
#

# MNIST input tensor ( with QueueRunner )
data = Mnist(batch_size=batch_size, num_epochs=num_epochs)
num_batch_per_epoch = data.train.num_batch


# input images and labels
x = data.train.image
y = data.train.label

# labels for discriminator
y_real = tf.ones(batch_size)
y_fake = tf.zeros(batch_size)


# discriminator labels ( half 1s, half 0s )
**y_disc = tf.concat(concat_dim=0, values=[y, y * 0])**

#
# create generator
#

# get random class number
z_cat = tf.multinomial(tf.ones((batch_size, cat_dim), dtype=tf.float32) / cat_dim, 1)
z_cat = tf.squeeze(z_cat, -1)
z_cat = tf.cast(z_cat, tf.int32)

# continuous latent variable
z_con = tf.random_normal((batch_size, con_dim))
z_rand = tf.random_normal((batch_size, rand_dim))

**z = tf.concat(concat_dim=1, values=[tf.one_hot(z_cat, depth = cat_dim), z_con, z_rand])**


# generator network
gen = generator(z)

# add image summary
# tf.sg_summary_image(gen)
tf.summary.image('real', x)
tf.summary.image('fake', gen)

#
# discriminator
disc_real, _, _ = discriminator(x)
disc_fake, cat_fake, con_fake = discriminator(gen)

# discriminator loss
**loss_d_r = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(logits=disc_real, targets=y_real))
loss_d_f = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(logits=disc_fake, targets=y_fake))**
loss_d = (loss_d_r + loss_d_f) / 2
print 'loss_d', loss_d.get_shape()
# generator loss
**loss_g = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(logits=disc_fake, targets=y_real))**

# categorical factor loss
loss_c = tf.reduce_mean(tf.nn.sparse_softmax_cross_entropy_with_logits(logits=cat_fake, labels=z_cat))

# continuous factor loss
loss_con =tf.reduce_mean(tf.square(con_fake-z_con))

train_disc, disc_global_step = optim(loss_d + loss_c + loss_con, lr=0.0001, optim = 'Adm', category='discriminator')
train_gen, gen_global_step = optim(loss_g + loss_c + loss_con, lr=0.001, optim = 'Adm', category='generator')
init = tf.global_variables_initializer()
saver = tf.train.Saver()
print train_gen

cur_epoch = 0
cur_step = 0


with tf.Session() as sess:
    sess.run(init)
    coord, threads = queue_context(sess)
    try:
        while not coord.should_stop():
            cur_step += 1
            dis_part = cur_step*1.0/num_batch_per_epoch
            dis_part = int(dis_part*50)
            sys.stdout.write("process bar ::|"+"<"* dis_part+'|'+str(cur_step*1.0/num_batch_per_epoch*100)+'%'+'\r')
            sys.stdout.flush()
            l_disc, _, l_d_step = sess.run([loss_d, train_disc, disc_global_step])
            l_gen, _, l_g_step = sess.run([loss_g, train_gen, gen_global_step])
            last_epoch = cur_epoch
            cur_epoch = l_d_step / num_batch_per_epoch
            if cur_epoch > max_epochs:
                break

            if cur_epoch> last_epoch:
                cur_step = 0
                print 'cur epoch {0} update l_d step {1}, loss_disc {2}, loss_gen {3}'.format(cur_epoch, l_d_step, l_disc, l_gen)
                if cur_epoch % save_epoch == 0:
                    # save
                    saver.save(sess, os.path.join('./checkpoint', 'info_gan'), global_step=l_d_step)
    except tf.errors.OutOfRangeError:
        print 'Train Finished'
    finally:
        coord.request_stop()


Thanks for your time!

**Comment by jetli6** [comment]
how much  are  your result ?

**Comment by Yrchuan** [comment]
I changed the code to LSGAN and it worked well, hope it help you : )
